conf/snowplow_monthlygood_checkout_event_schema_1_0_1.json

```diff
@@ -0,0 +1,39 @@
+{
+	"$schema": "https://raw.githubusercontent.com/kiva/snowplow/master/conf/snowplow_monthlygood_checkout_event_schema_1_0_1.json#",
+	"description": "Schema for Monthly Good checkout event",
+	"self": {
+		"vendor": "org.kiva",
+		"name": "monthlygood_checkout_event",
+		"format": "jsonschema",
+		"version": "1-0-1"
+	},
+
+	"type": "object",
+	"properties": {
+		"id": {
+			"type": "integer",
+			"description": "The auto deposit id"
+		},
+		"subscriptionType": {
+			"type": "string",
+			"description": "Subscription type: monthly, one-time"
+		},
+		"category": {
+			"type": "string",
+			"description": "Monthly Good category"
+		},
+		"depositAmount": {
+			"type": "number",
+			"description": "Monthly deposit amount (auto-deposit amount)"
+		},
+		"donationAmount": {
+			"type": "number",
+			"description": "Monthly donation amount"
+		},
+		"paymentType": {
+			"type": "string",
```

Comment:
Should this be an enum of strings? What are the expected values for the payment type?

Reply:
Well... where would we define this enum? For the Ui client, I was just going to pass whatever the BT drop-in presents for the payment type. Suppose we could agree on a map and document that somewhere...

Reply:
Enums can be defined [within the schema](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/mobile_context/jsonschema/1-0-1#L27-L30), but I'm ok with passing through the string from Braintree. The [names we're getting on mobile ](https://kiva.looker.com/sql/v5mtz6dgczsn2c?toggle=dat,vis) are below. Are they the same as the ones you're getting on the web?

```
Apple Pay
Visa
PayPal
Google Pay
MasterCard
AMEX
Discover
American Express
```

Reply:
Well that's a nice response... We're actually getting different values out of the Braintree payment type. PayPalAccount and CreditCard are what I'm seeing which are the only two options on Ui. This comes out of the braintreeResponse returned from the `braintreeCreateMonthlyGoodSubscription` mutation. Where are you all getting those values from...

@leoperry Does that sound right to you.

Reply:
We are mapping Braintree payment method data to an internal representation and use this "enum" class for the types: https://github.com/kiva/kiva/blob/development/libs/kv/lib/Braintree/PaymentMethod/PaymentMethodTypeEnum.php

This is funneled directly to the graphql enum (also `PaymentMethodTypeEnum`), but I'm not sure that's the source on the client side as that isn't part of the response currently. Maybe that value is coming over / through via the drop-in UI submission in some way?

Reply:
We get the payment method name in the nonce response from Braintree. The nonce object has a `type` string which is one of the values listed above. [iOS documentation](https://braintree.github.io/braintree_ios/Classes/BTPaymentMethodNonce.html#/c:objc(cs)BTPaymentMethodNonce(py)type), [Android documentation](https://www.javadoc.io/doc/com.braintreepayments.api/braintree/3.12.0/com/braintreepayments/api/models/PaymentMethodNonce.html)

Reply:
And should note we base our enums on Braintree's type enums: https://developers.braintreepayments.com/reference/response/payment-method-nonce/java#type

These names are generally shared throughout the server side SDKs across platforms, and it seems to overlap with the data in the JS payloads too: https://braintree.github.io/braintree-web-drop-in/docs/current/Dropin.html#~cardPaymentMethodPayload. I'd guess that this is where the string flowing through the drop-in UI on the web comes from.

Also noting that the (payment method) types outlined here are meant to differentiate between the different instrument types used for payment (CreditCard, PayPal, Venmo, AndroidPayCard, etc), whereas the type label strings in the iOS / Android documentation are meant to describe the payment (card) type and are more suitable to display to users (Visa, American Express, Apple Pay, etc). There may not be a huge difference in choice here as long as, e.g., instruments like PayPal are represented and we make a uniform choice. A differentiator here might be if we want to lump together, e.g., all `CreditCard`s (as in the backend SDK documentation) or we want to differentiate Visa, Amex, etc. (as in the iOS, Android documentation).

Reply:
Thanks @leoperry that's what we'd be passing along in the web event.